filebeat/docs/modules-dev-guide.asciidoc

```diff
@@ -0,0 +1,304 @@
+For the module name, you can us either a new module name or an existing module
+name. If the module doesn't exist, it will be created.
+
+In this guide we use `{fileset}` and `{module}` as placeholders for the fileset
+and module names. You need to replace these with the actual names you entered in
+the previous command.
+
+After running the above make command, you'll find the fileset, along with its
+generated files, under `module/{module}/{fileset}`. This directory
+contains the following files:
+
+[source,bash]
+----
+module/{module}/{fileset}
+├── manifest.yml
+├── config
+│   └── {fileset}.yml
+├── ingest
+│   └── pipeline.json
+├── _meta
+│   └── fields.yml
+└── test
+----
+
+Let's look at these files one by one.
+
+==== manifest.yml
+
+The `manifest.yml` is the control file for the module, where variables are
+defined and the other files are referenced. It is a YAML file, but in many
+places in it, built-in or defined variables can be used using the
+`{{.variable}}` syntax.
+
+The `var` section of the file defines the fileset variables and their default
+values. The module variables can be referenced in the rest of the configuration
+files, and their value can be overridden at runtime via the Filebeat
+configuration.
+
+As the fileset creator, you can use any names for the variables you define. Each
+variable must have a default value. So in it's simplest form, this is how you
+can define a new variable:
+
+[source,yaml]
+----
+var:
+  - name: pipeline
+    default: with_plugins
+----
+
+Most fileset should have a `paths` variable defined, in which the default paths
+where the log files are found should be set:
+
+[source,yaml]
+----
+var:
+  - name: paths
+    default:
+      - /example/test.log*
+    os.darwin:
+      - /usr/local/example/test.log*
+      - /example/test.log*
+    os.windows:
+      - c:/programdata/example/logs/test.log*
+----
+
+There's quite a lot going on in the above, so let's break it down:
+
+* The name of the variable is `paths` and the default value is
+  `["/example/test.log*"]`.
+* Note that variable values don't have to be strings,
+  they can be also numbers, objects, or like in this example, arrays.
+* We will use the `paths` variable to set the prospector
+  <<prospector-paths,paths>> setting, so "glob" values can be used here.
+* Besides the `default` value, the above also defines values for particular
+  operating systems: a default for darwin/OS X/macOS systems and a default for
+  Windows systems. These are introduced via the `os.darwin` and `os.windows`
+  keywords. The values under these keys become the default for the variable, if
+  Filebeat is executed on the respective OS.
+
+Besides the variable definition, the `manifest.yml` file also contains
+references to the ingest pipeline and prospector configuration to use (see next
+sections):
+
+[source,yaml]
+----
+ingest_pipeline: ingest/pipeline.json
+prospectors: config/testfileset.yml
```

Comment:
Just realising it now that this should probably be singular `prospector` as now only one prospector can be defined.